File: src/main/src/com/djulia/transactions/InMemoryAccountRepo.java
package com.djulia.transactions;

import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;

class InMemoryAccountRepo implements AccountRepo {
    private final Map<String, Account> accountStore = new ConcurrentHashMap<>();

    @Override
    public Optional<Account> findById(String accountId) {
        Account account = accountStore.get(accountId);
        if(account == null){
            return Optional.empty();
        }
        return Optional.of(account);
    }

    @Override
    public void save(Account account) {
        accountStore.put(account.getAccountNumber(), account);
    }

}

File: src/main/src/com/djulia/transactions/InMemoryAccountRepo.kt
package com.djulia.transactions

import java.util.Optional
import java.util.concurrent.ConcurrentHashMap

internal class InMemoryAccountRepo : AccountRepo {
    private val accountStore = ConcurrentHashMap<String, Account>()

    override fun findById(accountId: String): Account? {
        val account = accountStore[accountId] ?: return null
        return account
    }

    override fun save(account: Account) {
        accountStore.put(account.accountNumber, account)
    }

}
